src/plugins/explore/public/components/visualizations/visualization_container.tsx

```diff
@@ -72,13 +72,25 @@ export const VisualizationContainer = () => {
 
   const visualizationRegistry = useVisualizationRegistry();
 
-  // Initialize selectedChartType and its default styles when visualizationData changes
   useEffect(() => {
     if (visualizationData && visualizationData.visualizationType) {
-      dispatch(setSelectedChartType(visualizationData.visualizationType.type));
-      dispatch(setStyleOptions(visualizationData.visualizationType.ui.style.defaults));
+      if (!selectedChartType) {
+        // default initialization on load
+        dispatch(setSelectedChartType(visualizationData.visualizationType.type));
+        dispatch(setStyleOptions(visualizationData.visualizationType.ui.style.defaults));
+        return;
+      }
+      // if URL stores a different chart type than the visualizationData; we should persist the chart type from URL
+      if (selectedChartType !== visualizationData.visualizationType.type) {
+        const chartConfig = visualizationRegistry.getVisualizationConfig(selectedChartType);
+        setVisualizationData({
+          ...visualizationData,
+          visualizationType: chartConfig as VisualizationType<ChartType>,
```

Comment:
Hi there, it seems that the code here doesn't really persist chart type from URL but set the set the visualizationType in visualizationData with the state we stored in redux. I think we didn't really store the chart type in URL right?